### PR TITLE
Add Plugin Submission Documentation

### PR DESCRIPTION
**Pull request by MiguVT**
This PR adds a comprehensive guide to the Vencord documentation, detailing how users can submit plugins for official inclusion. The guide includes:

- Steps to fork and clone the Vencord repository.
- Instructions on adding a plugin to the `src/plugins` directory.
- How to commit changes and create a pull request.
- An overview of the review process and how to respond to feedback.
- Information about licensing and crediting plugin authors.

This guide aims to streamline the plugin submission process and provide clear instructions for contributors.

### CODE REVIEW

File: src/content/docs/plugins/submission.md [hunk 1]
```diff
@@ -2,3 +2,96 @@
...
+   ```bash
+   cd src/plugins
+   ```
+2. Create a new folder for your plugin using camelCase (e.g., `myFirstPlugin`):
+   ```bash
+   mkdir myFirstPlugin
+   ```
+3. Add your plugin files (`index.ts`, `README.md`, and optionally `native.ts`):
+    - **`index.ts`**: This is the main entry point for your plugin's code.
+    - **`README.md`**: Document your plugin, explain its functionality, and provide installation instructions and usage examples.
```
Comment (ImLvna) [inline src/content/docs/plugins/submission.md:41]: inbuilt plugins do not need installation instructions

File: src/content/docs/plugins/submission.md [hunk 2]
```diff
@@ -2,3 +2,96 @@
...
+    - **`native.ts`** (optional): If your plugin uses Node.js APIs, include this file.
+  
+   Make sure the `README.md` includes:
+   - A short description of your plugin.
+   - Instructions for installing and using the plugin.
```
Comment (ImLvna) [inline src/content/docs/plugins/submission.md:46]: we dont need installing

File: src/content/docs/plugins/submission.md [hunk 3]
```diff
@@ -2,3 +2,96 @@
...
+3. Ensure that the base repository is `Vendicated/Vencord` and the base branch is `main`.
+4. Provide a detailed description of your plugin, outlining what it does and any important information maintainers should know.
+
+### Step 5: Review Process
+
+After submitting the pull request, the Vencord maintainers will review your plugin. During this process:
+- You may receive feedback or requests for changes. Be responsive and adjust your plugin as needed.
+- The review ensures your plugin meets Vencord's standards for quality, security, and performance.
+
+Once the review is complete, your plugin will be merged into the `main` branch and included in future official Vencord releases.
```
Comment (ImLvna) [inline src/content/docs/plugins/submission.md:77]: plugins are first merged into dev

File: src/content/docs/plugins/submission.md [hunk 4]
```diff
@@ -2,3 +2,96 @@
...
+3. Push the changes to your forked repository:
+   ```bash
+   git push origin main
+   ```
+
+### Step 4: Create a Pull Request
+
+1. Go to your GitHub fork of the Vencord repository.
+2. Click on the **Pull Requests** tab and then click **New Pull Request**.
+3. Ensure that the base repository is `Vendicated/Vencord` and the base branch is `main`.
```
Comment (ImLvna) [inline src/content/docs/plugins/submission.md:68]: base branch should be dev, if it isnt then maintainers will switch it to dev anyway

**Review by ImLvna — CHANGES_REQUESTED** [review]

File: src/content/docs/plugins/submission.md [hunk 5]
```diff
@@ -2,3 +2,96 @@
...
+ */
+```
+
+If you prefer, you may change the `Vendicated and contributors` portion to your own name, but this is not strictly required as you will be included in the "contributors" group automatically.
+
+---
+
+By following this guide, you can submit your plugin for official inclusion in Vencord. Make sure your plugin is thoroughly tested and well-documented to streamline the review process and ensure its acceptance.
+
+For further assistance, feel free to ask for help in the [Vencord community channels](https://vencord.dev).
```
Comment (schoeneBiene) [inline src/content/docs/plugins/submission.md:97]: would probably make more sense to have this link to the discord

File: src/content/docs/plugins/submission.md [hunk 6]
```diff
@@ -2,3 +2,96 @@
...
+
+1. Navigate to the `src/plugins` folder in your cloned repository:
+   ```bash
+   cd src/plugins
+   ```
+2. Create a new folder for your plugin using camelCase (e.g., `myFirstPlugin`):
+   ```bash
+   mkdir myFirstPlugin
+   ```
+3. Add your plugin files (`index.ts`, `README.md`, and optionally `native.ts`):
```
Comment (schoeneBiene) [inline src/content/docs/plugins/submission.md:39]: it doesn't make sense to list these (other than the readme, but that should probably be a seperate point) files, as these would already have been created before this

**Review by schoeneBiene — COMMENTED** [review]
would probably be better to add a part on feature branches

File: src/content/docs/plugins/submission.md (hunk 1 above)
Comment (MiguVT) [reply src/content/docs/plugins/submission.md]: Done, resolved everything in last commit!

File: src/content/docs/plugins/submission.md (hunk 2 above)
Comment (MiguVT) [reply src/content/docs/plugins/submission.md]: Done, resolved everything in last commit!

File: src/content/docs/plugins/submission.md (hunk 3 above)
Comment (MiguVT) [reply src/content/docs/plugins/submission.md]: Done, resolved everything in last commit!

File: src/content/docs/plugins/submission.md (hunk 5 above)
Comment (MiguVT) [reply src/content/docs/plugins/submission.md]: Done, resolved everything in last commit!

File: src/content/docs/plugins/submission.md (hunk 6 above)
Comment (MiguVT) [reply src/content/docs/plugins/submission.md]: Done, resolved everything in last commit!

**Comment by MiguVT** [comment]
Done, resolved everything in last commit! tysm for your time

File: src/content/docs/plugins/submission.md [hunk 7]
```diff
@@ -36,16 +38,12 @@ The submission process involves forking the Vencord repository, adding your plug
...
-    - **`native.ts`** (optional): If your plugin uses Node.js APIs, include this file.
-  
-   Make sure the `README.md` includes:
-   - A short description of your plugin.
-   - Instructions for installing and using the plugin.
+3. Ensure your `README.md` file is added and includes:
+   - A brief description of your plugin.
    - Screenshots, GIFs, or videos if applicable to showcase your plugin.
 
+> Since this is for a built-in plugin, no installation instructions are required.
```
Comment (ImLvna) [inline src/content/docs/plugins/submission.md:45]: please dont use chatgpt for docs 😭

File: src/content/docs/plugins/submission.md [hunk 8]
```diff
@@ -65,16 +63,16 @@ The submission process involves forking the Vencord repository, adding your plug
 
 1. Go to your GitHub fork of the Vencord repository.
 2. Click on the **Pull Requests** tab and then click **New Pull Request**.
-3. Ensure that the base repository is `Vendicated/Vencord` and the base branch is `main`.
-4. Provide a detailed description of your plugin, outlining what it does and any important information maintainers should know.
+3. Ensure the base repository is `Vendicated/Vencord` and the base branch is `dev`. If you select `main`, maintainers will switch it to `dev` anyway.
```
Comment (ImLvna) [inline src/content/docs/plugins/submission.md:66]: did you literally pipe our comments into chatgpt and tell it to fix it

**Comment by Vendicated** [comment]
thanks for trying to help out, but this is
1) not really what I intended for this page
2) incorrect in multiple aspects

File: src/content/docs/plugins/submission.md (hunk 7 above)
Comment (MiguVT) [reply src/content/docs/plugins/submission.md:45]: I used it for fix it fast, as i dont had time rn, if want something better i can do it but first i should have more time to make it correct and dont use AI for corrections

**Pull request closed without merging**